src/richie/apps/core/cache.py

```diff
@@ -131,3 +135,40 @@ def close(self, **kwargs):
         del self._client_read
         self._client_write = None
         self._client_read = None
+
+
+class LimitBrowserCacheTTLHeaders(MiddlewareMixin):
+    """
+    This middleware allows to define a maximum cache timeout for the cache
+    response headers (Cache-control: max-age and Expires).
+
+    It can be configured with the MAX_BROWSER_CACHE_TTL setting.
+    This value is expressed in seconds.
+
+    LimitBrowserCacheTTLHeaders should be placed at the beginning of the
+    MIDDLEWARE list, so that it'll get called last during the response phase.
+    """
+
+    # pylint: disable=no-self-use
+    def process_response(self, request, response):
+        """
+        Rewrite the "Cache-control" and "Expires headers" in the response
+        if needed.
+        """
+        try:
+            max_ttl = int(getattr(settings, "MAX_BROWSER_CACHE_TTL", None))
+            if max_ttl < 0:
+                raise ValueError("MAX_BROWSER_CACHE_TTL must be a positive integer")
```

Comment:
@madmatah I'm now wondering if it would not have been better to use 600 as default when it is not defined in settings...

Reply:
IMO this is an opinionated default value. It should be set with full knowledge of the facts.
Maybe, doing nothing silently would have been better ?

Reply:
I think no because you would think that it works. My point of view was that it is no so much opionated as it is a CMS : for everyone, it is the time elapsed between a publication and the moment everyone sees it for sure. 10 minutes seemed a good default for that?